torch/quantization/fx/_equalize.py

```diff
@@ -244,11 +249,54 @@ def get_op_node_and_weight_eq_obs(
         return op_node, weight_eq_obs
 
     elif op_node.op == 'call_function':
-        # TODO
-        return None, None
+        weight_node = maybe_get_functional_weight_node(op_node, modules, 'eq_obs')
+        if weight_node is not None:
+            weight_eq_obs = modules[str(weight_node.target)]
+            assert(isinstance(weight_eq_obs, _WeightEqualizationObserver))
+            return op_node, weight_eq_obs
 
     return None, None
 
+def maybe_get_functional_weight_node(op_node: Node, modules: Dict[str, nn.Module], attr: str) -> Optional[Node]:
+    """ Retrieves the weight node with the given attribute that is input into
+    the given op_node.
+
+    The path should look something like: get_attr(weight) -> weight_quant_obs -> weight_eq_obs -> op_node
+    So we can trace back a specific number of steps to find the node with the
+    given attribute.
+
+    Args:
+        op_node: A functional node
+        attr: Can be one of {'eq_obs', 'quant_obs', 'get_attr'}. Specifies the
+              node we are looking for which will be used to find the number of
+              steps to take from the op_node before returning the node
+    """
+    WEIGHT_ATTR_INDEX_DICT = {'eq_obs': 1, 'quant_obs': 2, 'get_attr': 3}
+
+    # Get the weight equalization observer
+    node = None
+    assert(op_node.op == 'call_function' and op_node.target in WEIGHT_INDEX_DICT)
+    for i, node_arg in enumerate(op_node.args):
+        if i in WEIGHT_INDEX_DICT[op_node.target]:  # type: ignore[index]
+            node = node_arg
+            break
+    if node is None:
+        raise LookupError('Could not find the equalization observer')
+
+    assert(isinstance(node, Node) and node.op == 'call_module' and
+           isinstance(modules[str(node.target)], _WeightEqualizationObserver))
```

Comment:
can this part be factored into one function that gets the weight_equalization observer? or is there already such a function